### DISCUSSION TITLE
Clear cart after checkout

### DISCUSSION THREAD

**Discussion by sattyframework** [duncanmcclean/simple-commerce#908] [Help] [answered]
Hello I am a bit stuck. How can i clear the cart after the payment has been successful? 

I am handling the payment through a controller and I am using database orders. I know that I can use events e.g.  postCheckout however I am using a custom model for my orders instead of the simple commerce OrderModel::class and it does not accept my Model type. 

Any help is appreciated! Thanks

**Comment by duncanmcclean** [comment]
Are you using your own controller?

**Reply by sattyframework** [reply]
Yes! The checkout process in in vue as it is a multi-step which then posts to my controller to handle the payment. All is working, and just need to perform some post checkout steps such as clearing the current the cart :)

**Reply by duncanmcclean** [reply, marked as answer]
Ah okay, I don't generally recommend using custom controllers.

Anyways, this is where the cart is cleared in Simple Commerce's `CheckoutController`: https://github.com/duncanmcclean/simple-commerce/blob/f16adef59c665da43b457252e2fbc69a5c28a857/src/Http/Controllers/CheckoutController.php#L278

You'll need to import the `CartDriver` trait.
